packages/ui/Touchable/PinchZoom.js

```diff
@@ -0,0 +1,177 @@
+import clamp from 'ramda/src/clamp';
+import PropTypes from 'prop-types';
+
+class PinchZoom {
+	pinchZoomConfig = null;
+	startScale = 1.0;
+	scale = 1.0;
+
+	isZooming = () => this.pinchZoomConfig != null;
+
+	setContainerBounds = (node) => {
+		const {global: isGlobal, boxSizing} = this.pinchZoomConfig;
+		let bounds = null;
+
+		if (typeof window === 'undefined' || !node) return;
+
+		if (isGlobal) {
+			bounds = {
+				minX: 0,
+				minY: 0,
+				maxX: window.innerWidth,
+				maxY: window.innerHeight
+			};
+		} else {
+			bounds = node.getBoundingClientRect();
+
+			// adjust for padding when using content-box
+			if (boxSizing === 'content-box') {
+				const computedStyle = window.getComputedStyle(node);
+				bounds = {
+					minX: bounds.left + parseInt(computedStyle.paddingLeft),
+					minY: bounds.top + parseInt(computedStyle.paddingTop),
+					maxX: bounds.right - parseInt(computedStyle.paddingRight),
+					maxY: bounds.bottom - parseInt(computedStyle.paddingBottom)
+				};
+			} else {
+				bounds = {
+					minX: bounds.left,
+					minY: bounds.top,
+					maxX: bounds.right,
+					maxY: bounds.bottom
+				};
+			}
+		}
+
+		this.bounds = bounds;
+	};
+
+	getBoundsCoords = ({x, y}) => {
+		const {maxX, maxY, minX, minY} = this.bounds;
+
+		return {x: clamp(minX, maxX, x) - minX, y: clamp(minY, maxY, y) - minY};
+	};
+
+	updateZoom = (scale) => {
+		const {maxZoom, minZoom} = this.pinchZoomConfig;
+		const newScale = clamp(minZoom, maxZoom, scale);
+
+		if (newScale !== this.scale) {
+			this.scale = newScale;
+			return true;
+		}
+
+		return false;
+	};
+
+	begin = (config, {noResume, onPinchZoom, onPinchZoomEnd, onPinchZoomStart}, coords, node) => {
+		if (!onPinchZoom && !onPinchZoomStart && !onPinchZoomEnd) {
+			return;
+		}
+
+		this.pinchZoomConfig = {
+			...config,
+			node,
+			resume: !noResume
+		};
+
+		this.setContainerBounds(node);
+		const {x: x1, y: y1} = this.getBoundsCoords(coords[0]);
```

Comment:
Not a strong suggestion...
If `getBoundsCoords` returns an array rather than an object, this line could be the following;

> const [x1, y1] = this.getBoundsCoords(coords[0]);

Reply:
I made getDistance func